src/quill.imageCompressor.ts

```diff
@@ -100,23 +86,29 @@ class imageCompressor {
     return dataUrlCompressed;
   }
 
-  insertToEditor(url: string, blob: Blob) {
-    if (this.options.insertIntoEditor) {
-      this.options.insertIntoEditor(url, blob, this.quill);
-    } else {
-      this.Logger.log('insertToEditor', {url});
-      this.range = this.quill.getSelection();
-      const range = this.range;
-      if (!range) {
-        return;
-      }
-      // Insert the compressed image
-      this.logFileSize(url);
-      this.quill.insertEmbed(range.index, "image", `${url}`, "user");
-      // Move cursor to next position
-      range.index++;
-      this.quill.setSelection(range, "api");
+  async processImage(dataUrl: string): Promise<string> {
+    dataUrl = await this.downscaleImageFromUrl(dataUrl);
+    if (this.options.uploadImage) {
+      dataUrl = await this.options.uploadImage(b64toBlob(dataUrl));
```

Comment:
I find this confusing, it's modifying the `dataUrl` argument in several ways and also returning it 😕